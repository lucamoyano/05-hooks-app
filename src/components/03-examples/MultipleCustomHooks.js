import React from 'react'
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';

import '../../index.css';

export const MultipleCustomHooks = () => {

    const { counter, increment } = useCounter(1);
    
    const { loading, data } = useFetch(`https://breakingbadapi.com/api/quotes/${counter}`);

    //Si data!= null, extrae la informaci√≥n de data[0]
    const { author, quote } = !!data && data[0] 


    return (
        <>
            <h1>BreakingBad Quotes</h1>
            <hr/>
            {
                loading 
                ?
                    (
                        <div className="alert alert-info text-center">
                            Loading...
                        </div>
                    )
                :
                    (
                        <blockquote className="blockquote text-end">
                            <p className="mb-2"> { quote } </p>
                            <footer className="blockquote-footer"> { author } </footer>
                        </blockquote>
                    )
            }

            <button 
                onClick={ increment }
                className="btn btn-primary"
            >Siguiente frase
            </button>

        </>
    )
}
