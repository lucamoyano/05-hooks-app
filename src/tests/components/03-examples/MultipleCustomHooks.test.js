import { shallow }  from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';

jest.mock('../../../hooks/useFetch'); //Simular llamado a funcion en componente
jest.mock('../../../hooks/useCounter'); 


describe('Pruebas en <MultipleCustomHooks />',()=> {
    
    beforeEach( () => {
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        })
    });
    
    test('debe de mostrarse correctmente', () => {

        //Valor por defecto del useFetch
        useFetch.mockReturnValue({
            data:null,
            loading:false,
            error:null
        });

        const wrapper = shallow(<MultipleCustomHooks /> );
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar la informaciÃ³n', () => {

        useFetch.mockReturnValue({
            data:[{
                author: 'Lucas',
                quote: 'Hola mundo'
            }],
            loading:false,
            error:null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.alert').exists()).toBe(false);

    });
    
    

})