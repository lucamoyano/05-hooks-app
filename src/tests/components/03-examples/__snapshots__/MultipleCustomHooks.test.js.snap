// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctmente 1`] = `
<Fragment>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <blockquote
    className="blockquote text-end"
  >
    <p
      className="mb-2"
    >
       
       
    </p>
    <footer
      className="blockquote-footer"
    >
       
       
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</Fragment>
`;
